### PR TITLE
More usage-friendly and explicitly-handled error and info printouts for 'rauc status' (subcommands)

### PR DESCRIPTION
**Pull request by ejoerns**
Calling 'rauc status' from the command line still produced some noise and used logging methods instead of `g_print()`/`g_printerr()` as we otherwise use halfway consistently in the CLI frontend.

We also do not need to call the boot selection backend when only invoking `mark-*` subcommands.

### CODE REVIEW

**Comment by ejoerns** [comment]
Rebased onto master to fix conflict in `src/main.c`.

**Comment by jluebbe** [comment]
@ejoerns the status handler function is becoming somewhat hard to follow. Perhaps we should split printing and mark-* into separate functions?

**Comment by jluebbe** [comment]
I've dropped the last commit from this PR, so we can get this merged sooner and discuss it separately.

**Comment by ejoerns** [comment]
Looks fine for me! :+1:
